tests/helpers.py

```diff
@@ -338,3 +338,32 @@ def redirect_request_callback(
             status=status_code,
             body=redirect_request_callback,
         )
+
+
+@contextlib.contextmanager
+def switch_working_directory(path: Path, remove: bool = False) -> Iterator[Path]:
```

Comment:
How about using the existing code [here](https://github.com/python-poetry/poetry/blob/64d8931538baaf91e4021ad245389c544e6e788c/src/poetry/utils/helpers.py#L65) instead of introducing new one?

Reply:
The functionality is different, and the "remove" use case is only required for the test suite.

Reply:
ok, now the question remains, where is a test for a test helper? `test_test_helpers.py` maybe?

Reply:
We can potentially deal with in a separate issue? But I feel like that is overkill since the code is not overly complex.